src/scene/Scene.ts

```diff
@@ -153,9 +153,9 @@ class Scene {
 
   insertElementAtIndex(element: ExcalidrawElement, index: number) {
     const nextElements = [
-      ...this.elements.slice(0, index + 1),
+      ...this.elements.slice(0, index),
       element,
-      ...this.elements.slice(index + 1),
+      ...this.elements.slice(index),
```

Comment:
btw, this now means that if ever call this function as you did before the change, passing a result directly from `findIndex`, it will not work. So I'd add a check for `index > -1` and probably throw (so we can discover the bug instead of getting an undefined behavior).

Reply:
dpme/